### PR TITLE
Fixes #5176 - Adding vSphere as an OS type

### PR DESCRIPTION
**Pull request by LiamLeane**
This was originally theforeman/foreman#3351, github isn't opening the pull back up after clearing commits and submitting a new one so new PR.

Adds vSphere as an OS type for VCSA, ESXi etc.

### CODE REVIEW

**Comment by timogoebel** [comment]
@theforeman-bot ok to test

**Comment by timogoebel** [comment]
@LiamLeane : First of all, thanks for this PR. I think, every new os is a great addition to Foreman.

I didn't have time to take a closer look, but you definitely need to fix the test in `test/models/operatingsystem_test.rb:162`.

**Comment by dLobatog** [comment]
Seems OK, thanks @LiamLeane ! What templates are you using to deploy this? The ones in the Foreman wiki? http://projects.theforeman.org/projects/foreman/wiki/VMware_ESXi ? 

If so - would you mind to share the ones you're using? I'd like to deploy ESXi using this and merge. Thanks!

**Comment by timogoebel** [comment]
@dLobatog : There is a PR that adds the templates: https://github.com/theforeman/community-templates/pull/249

File: app/models/operatingsystems/vsphere.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  class << self
+    delegate :model_name, :to => :superclass
+  end
+
+  def mediumpath(host)
+    medium_uri(host).to_s
+  end
+
+  def pxe_type
```
Comment (dLobatog) [inline app/models/operatingsystems/vsphere.rb:12]: This needs also 

``` ruby
def pxedir
  "EFI/BOOT"
end
```

Not sure 100% about whether that path suits all versions but for 6.0 that seems OK

File: app/models/operatingsystems/vsphere.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    "kickstart"
+  end
+
```
Comment (dLobatog) [inline app/models/operatingsystems/vsphere.rb:15]: This needs an `available_loaders` method too to list all of the possible PXE loaders

**Review by dLobatog — CHANGES_REQUESTED** [review]

**Pull request closed without merging**